src/app-layer-ssl.c

```diff
@@ -378,30 +543,186 @@ static int TLSDecodeHandshakeHello(SSLState *ssl_state, uint8_t *input,
                 ssl_state->curr_connp->sni = SCMalloc(sni_strlen);
 
                 if (unlikely(ssl_state->curr_connp->sni == NULL))
-                    goto end;
+                    return -1;
 
-                memcpy(ssl_state->curr_connp->sni, input, sni_strlen - 1);
+                memcpy(ssl_state->curr_connp->sni, *input, sni_strlen - 1);
                 ssl_state->curr_connp->sni[sni_strlen-1] = 0;
 
-                input += sni_len;
+                *input += sni_len;
+                break;
+            }
+
+            case SSL_EXTENSION_ELLIPTIC_CURVES:
```

Comment:
I thnk it would make sense to break up this function/loop as it's quite long

Reply:
I'll create some new functions for you :)